gui-wm/tinywl/tinywl-0.17.3.ebuild

```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="\"minimum viable product\" Wayland compositor based on wlroots"
+HOMEPAGE="https://gitlab.freedesktop.org/wlroots/wlroots"
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://gitlab.freedesktop.org/wlroots/wlroots.git"
+	inherit git-r3
+else
+	SRC_URI="https://gitlab.freedesktop.org/wlroots/wlroots/-/releases/${PV}/downloads/wlroots-${PV}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/wlroots-${PV}"
+fi
+
+LICENSE="CC0-1.0"
+SLOT="0"
+DEPEND="
+	dev-libs/wayland
+	x11-libs/libxkbcommon
+	=gui-libs/wlroots-$(ver_cut 1-2)*:=
+"
+RDEPEND="
+	${DEPEND}
+	!!gui-libs/wlroots[tinywl(-)]
```

Comment:
A hard blocker isn't right here and it prevents the dependency being resolved cleanly by Portage (it requires manual intervention). Hard blockers are only needed if something can't even be built if the other thing is installed, as opposed to a file collision which "allows" a file collision.